libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/PolicyManager.java

```diff
@@ -59,6 +59,68 @@
 import static java.util.zip.ZipFile.OPEN_READ;
 import static org.elasticsearch.entitlement.bridge.Util.NO_CLASS;
 
+/**
+ * This class is responsible for finding the "layer" (system, server, plugin, agent) for a caller class to check, retrieve the policy
```

Comment:
We seem to be mixing two meanings of the word "layer" in this passage:
- A module layer
- An "entitlements layer" that seems to correspond to what we currently call a "component", plus there's another one for system modules.

The system and server "entitlement layers" are both in the boot "module layer". This seems kind of confusing.

Also, very soon, this class will no longer be responsible for finding the component, because `ScopeResolver` will do that.